model_analyzer/result/run_config_measurement.py

```diff
@@ -0,0 +1,373 @@
+
+        return [
+            model_config_measurement.non_gpu_data()
+            for model_config_measurement in self._model_config_measurements
+        ]
+
+    def get_metric(self, tag):
+        """
+        Parameters
+        ----------
+        tag : str
+            A human readable tag that corresponds
+            to a particular metric
+
+        Returns
+        -------
+        per model list of Records
+            metric Record corresponding to
+            the tag, in this measurement, None
+            if tag not found.
+        """
+        return [
+            model_config_measurement.get_metric(tag)
+            for model_config_measurement in self._model_config_measurements
+        ]
+
+    def get_weighted_metric(self, tag):
+        """
+        Parameters
+        ----------
+        tag : str
+            A human readable tag that corresponds
+            to a particular metric
+
+        Returns
+        -------
+        per model list of Records
+            weighted metric Record corresponding to
+            the tag, in this measurement, None
+            if tag not found.
+        """
+        return [
+            model_config_measurement.get_metric(tag) *
+            self._model_config_weights[index]
+            for index, model_config_measurement in enumerate(
+                self._model_config_measurements)
+        ]
+
+    def get_metric_value(self, tag, default_value=0):
+        """
+        Parameters
+        ----------
+        tag : str
+            A human readable tag that corresponds
+            to a particular metric
+        default_value : any
+            Value to return if tag is not found
+
+        Returns
+        -------
+        list of Records
+            Average of the values of the metric Record corresponding 
+            to the tag, default_value if tag not found.
+        """
+        return mean([
+            default_value if m is None else m.value()
+            for m in self.get_metric(tag)
+        ])
+
+    def get_weighted_metric_value(self, tag, default_value=0):
+        """
+        Parameters
+        ----------
+        tag : str
+            A human readable tag that corresponds
+            to a particular metric
+        default_value : any
+            Value to return if tag is not found
+
+        Returns
+        -------
+        list of Records
+            Weighted average of the values of the metric Record corresponding 
+            to the tag, default_value if tag not found.
+        """
+        weighted_sum = 0
+        for index, metric in enumerate(self.get_metric(tag)):
+            weighted_sum += default_value if metric is None else metric.value(
+            ) * self._model_config_weights[index]
+
+        return weighted_sum
+
+    def gpus_used(self):
+        """
+        Returns
+        -------
+        list of ints
+            list of device IDs used in this measurement
+        """
+
+        return list(self._gpu_data.keys())
```

Comment:
will this ever return the same gpu multiple times?  Should this be a `set`?

Reply:
Dictionaries keys must be unique, so this is guaranteed to be unique